src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,168 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.NoSuchElementException;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node head;
+    private Node tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            head = new Node<>(null, value, null);
+            tail = head;
+        } else {
+            Node<T> node = new Node<>(tail, value, null);
+            tail.next = node;
+            tail = node;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("");
+        }
+
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> nodeAtIndex = findNodeByIndex(index);
+            Node<T> currentNode = new Node<>(nodeAtIndex.prev, value, nodeAtIndex);
+
+            if (nodeAtIndex.prev != null) {
+                nodeAtIndex.prev.next = currentNode;
+            } else {
+                head = currentNode;
+            }
+            nodeAtIndex.prev = currentNode;
+            size++;
+        }
```

Comment:
```suggestion
        if (index == size) {
            add(value);
            return;
        }
        
        Node<T> nodeAtIndex = findNodeByIndex(index);
        Node<T> currentNode = new Node<>(nodeAtIndex.prev, value, nodeAtIndex);

        if (nodeAtIndex.prev != null) {
            nodeAtIndex.prev.next = currentNode;
        } else {
            head = currentNode;
        }
        nodeAtIndex.prev = currentNode;
        size++;
```
indented, nested code is harder to read, let's just return early
[see this](softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement)